test/functional/test_framework/wallet.py

```diff
@@ -125,16 +125,18 @@ def _bulk_tx(self, tx, target_weight):
         """Pad a transaction with extra outputs until it reaches a target weight (or higher
         at most BULK_TX_PADDING_OFFSET WU).
         """
-        tx.vout.append(CTxOut(nValue=0, scriptPubKey=CScript([OP_RETURN])))
-        # determine number of needed padding bytes by converting weight difference to vbytes
-        dummy_vbytes = (target_weight - tx.get_weight() + BULK_TX_PADDING_OFFSET) // WITNESS_SCALE_FACTOR
-        # compensate for the increase of the compact-size encoded script length
-        # (note that the length encoding of the unpadded output script needs one byte)
-        dummy_vbytes -= len(ser_compact_size(dummy_vbytes)) - 1
-        tx.vout[-1].scriptPubKey = CScript([OP_RETURN] + [OP_1] * dummy_vbytes)
-        # Actual weight should be at most BULK_TX_PADDING_OFFSET higher than target weight
-        assert_greater_than_or_equal(tx.get_weight(), target_weight)
-        assert_greater_than_or_equal(target_weight + BULK_TX_PADDING_OFFSET, tx.get_weight())
+        # Prevent bulking tx when target weight is below the sum of tx weight and BULK_TX_PADDING_OFFSET
+        if (target_weight + BULK_TX_PADDING_OFFSET) > tx.get_weight():
```

Comment:
The condition seems correct but the corresponding comment doesn't, it was a bit difficult for me to understand.

How about the following?
```
Prevent bulking tx when target weight (along with offset) is less than or equal to the tx weight
```

Reply:
Yes, updated.